sections/sets.tex

```diff
@@ -183,13 +183,35 @@ \subsection{Natural numbers}
   \end{prooftree}
 \]
 
-The recursion symbol introduced by the elimination rule has a simple semantical
-explanation in terms of its computations: it first evaluates $n$, which is known
-to yield either $0$ or $s(n)$ for some $\inj{n}{\NN}$. In the first case, it
-evaluates $d$ and returns the result. Otherwise, it takes $n$ and feeds it to
-$e$ as its first argument, together with the result of evaluating itself on the
-predecessor, namely $n$. The second elimination rule shows that the recursor
-respects equality in the obvious way:
+$\natrec{n}{d}{(x,y)e(x,y)}$ in the conclusion has the following semantics\footnote{Here, the
+  notation $(x,y)e(x,y)$ denotes an abstraction of $(x,y)$ over a term
+  $e$ where $(x,y)$ can possibly appear free; those with a background
+  in (functional) programming or $\lambda$-calculus might think of
+  $\lambda(x,y).e$.
+
+  The recursion symbol $\natrec{c}{d}{(x,y)e(x,y)}$ is generally
```

Comment:
Non mi convince, per due motivi:

- "generally understood" sembra che prendiamo il simbolo R e il suo significato ispirandoci alla letteratura esistente, dove esso e` solitamente definito bla bla. In realta` R e` una costante che stiamo definendo noi **ora**, la cui semantica non e` generally understood ma **completely understood** perche` la definiamo noi ora e con precisione. La spiegazione semantica di R infatti merita piu` di una footnode. In realta` il suo significato operazionale "per casi" e` dato nell'elenco puntato due righe dopo. E` davvero necessario ripetere la stessa cosa in una footnote?